# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/url-classifier/Classifier.h
# Commit: d49cd2aa4568
# Full Hash: d49cd2aa45686212954493e2465fc3258c1796a2
# Author: dlee <dlee@mozilla.com>
# Date: 2019-08-21 16:47:43
# Regressor Bug: 1562822
# File Overlap Count: 1
# Description:
#   Bug 1562822 - P2. Reset corrupted Safe Browsing database before triggering an update. r=gcp
#   
#   Patch P2 & P3 refine how Safe Browsing handles Safe Browsing database
#   loading failure.
#   
# ==============================================================================

diff -r 7a0ab3a7558b -r d49cd2aa4568 toolkit/components/url-classifier/Classifier.h
--- a/toolkit/components/url-classifier/Classifier.h	Wed Aug 21 12:07:44 2019 +0000
+++ b/toolkit/components/url-classifier/Classifier.h	Wed Aug 21 12:08:03 2019 +0000
@@ -151,7 +151,9 @@
   nsresult DumpFailedUpdate();
 #endif
 
-  nsresult ScanStoreDir(nsIFile* aDirectory, nsTArray<nsCString>& aTables);
+  nsresult ScanStoreDir(nsIFile* aDirectory,
+                        const nsTArray<nsCString>& aExtensions,
+                        nsTArray<nsCString>& aTables);
 
   nsresult UpdateHashStore(TableUpdateArray& aUpdates,
                            const nsACString& aTable);
@@ -170,7 +172,11 @@
 
   bool CheckValidUpdate(TableUpdateArray& aUpdates, const nsACString& aTable);
 
-  nsresult LoadMetadata(nsIFile* aDirectory, nsACString& aResult);
+  nsresult LoadHashStore(nsIFile* aDirectory, nsACString& aResult,
+                         nsTArray<nsCString>& aFailedTableNames);
+
+  nsresult LoadMetadata(nsIFile* aDirectory, nsACString& aResult,
+                        nsTArray<nsCString>& aFailedTableNames);
 
   static nsCString GetProvider(const nsACString& aTableName);
 